.github/workflows/test.yml

```diff
@@ -96,3 +96,53 @@ jobs:
         run: |
           call "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" x64
           bash -c "ci/scripts/build_example.sh $(pwd)/example"
+  ubuntu-meson:
+    name: Meson - AMD64 Ubuntu 24.04
+    runs-on: ubuntu-24.04
+    timeout-minutes: 30
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
+      - name: Checkout iceberg-cpp
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          fetch-depth: 0
+      - name: Install build dependencies
+        run: |
+          python3 -m pip install --upgrade pip
+          python3 -m pip install meson ninja
+      - name: Build Iceberg
```

Comment:
These CI jobs just invoke Meson directly, as I think the steps are pretty straightforward. However, some of the other Apache projects choose to still put this all into the `ci/build_xxx.sh` script - happy to migrate there if preferable